packages/dbml-core/src/parse/connection/postgres/utils.js

```diff
@@ -0,0 +1,519 @@
+/* eslint-disable camelcase */
+import { Client } from 'pg';
+import {
+  Endpoint,
+  Enum,
+  Field,
+  Index,
+  Table,
+  Ref,
+} from '../../ANTLR/ASTGeneration/AST';
+
+const connectPg = async (connection) => {
+  const client = new Client(connection);
```

Comment:
When trying to connect without providing a password, I encounter this error, and it hang the process for me.

![Screenshot from 2024-07-31 11-45-58](https://github.com/user-attachments/assets/55ca47be-1f72-4257-82e8-aa999774bc11)

I think this is an uncatchable error in pg and you should provide the connection arguments a default password to avoid this error.

Reply:
We switched to use connection string => we do not care about this issue again.